shared/actions/profile/proofs.js

```diff
@@ -100,26 +99,31 @@ function* addServiceProof(_, action) {
   })
 
   const checkProofTask = yield Saga._fork(function*() {
-    yield Saga.take(ProfileGen.checkProof)
-    if (_outputInstructionsResponse) {
-      _outputInstructionsResponse.result()
-      _outputInstructionsResponse = null
+    while (true) {
+      yield Saga.take(ProfileGen.checkProof)
+      if (_outputInstructionsResponse) {
+        _outputInstructionsResponse.result()
+        _outputInstructionsResponse = null
+      }
     }
   })
 
   const submitUsernameTask = yield Saga._fork(function*() {
-    yield Saga.take(ProfileGen.submitUsername)
-    yield Saga.put(ProfileGen.createCleanupUsername())
-    if (_promptUsernameResponse) {
-      yield Saga.put(
-        ProfileGen.createUpdateErrorText({
-          errorCode: null,
-          errorText: '',
-        })
-      )
-      const state = yield* Saga.selectState()
-      _promptUsernameResponse.result(state.profile.username)
-      _promptUsernameResponse = null
+    // loop since if we get errors we can get these events multiple times
+    while (true) {
```

Comment:
the lack of this while loop causes this to not work in master. if you enter a bad web name and then try and continue it'll never keep moving forward as our take has died. this keeps it going while the rpc is still alive